### ISSUE TITLE
Add support for composite keys

### ISSUE THREAD

**Issue by adamskt** [IntelliTect/Coalesce#103]
Because the ClassViewModel for model entities only takes the [first property](https://github.com/IntelliTect/Coalesce/blob/320c376bb4751ed488708fab8fc66291c3a208c3/src/IntelliTect.Coalesce/TypeDefinition/ClassViewModel.cs#L367) with IsPrimaryKey and additionally because the IsPrimaryKey property only looks for the KeyAttribute (ignoring anything set up in the [fluent interface](https://docs.microsoft.com/en-us/ef/core/modeling/keys#fluent-api)), composite keys aren't currently supported by the generated API controllers.

Stuff needed to support this.

- [ ] Change ClassViewModel to have a collection of primary keys,
- [ ] Change ApiController generator to generate anonymous types for key parameters when keys number more than one.
- [ ] Change IsPrimaryKey impl to be aware of both fluently-defined keys and attributed ones.

**Comment by GrantErickson** [comment]
There are also several spots where the key is returned as a query string. We would need to figure this one out as well. Significant bit of work here. Not sure if the complexity is worth it rather than just adding a single primary key which some would hold is a good practice anyway.

**Comment by adamskt** [comment]
I agree it's non-trivial and complex, but it would also enable the multi-tenant #5 story with SQL row-level security (i.e. a tenant_id field on every model).

**Comment by GrantErickson** [comment]
Yes, that is true. I think we could also have it along side a unique single column key. Which might be easier to generate keys for as well.

**Comment by ascott18** [comment]
Probably dependent on #266 to make the DX nice
